ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/ViolationsByPolicySeverity.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { gql, useQuery } from '@apollo/client';
+import { Flex, FlexItem, Title, Button, Divider } from '@patternfly/react-core';
+
+import LinkShim from 'Components/PatternFly/LinkShim';
+import useURLSearch from 'hooks/useURLSearch';
+import { violationsBasePath } from 'routePaths';
+import { SearchFilter } from 'types/search';
+import { DeploymentAlert } from 'types/alert.proto';
+import { getQueryString } from 'utils/queryStringUtils';
+import { getRequestQueryStringForSearchFilter } from 'utils/searchUtils';
+
+import { severities } from 'constants/severities';
+import WidgetCard from './WidgetCard';
+import MostRecentViolations from './MostRecentViolations';
+import PolicyViolationTiles from './PolicyViolationTiles';
+import useAlertGroups from '../hooks/useAlertGroups';
+
+function getViewAllLink(searchFilter: SearchFilter) {
+    const queryString = getQueryString({
+        s: {
+            ...searchFilter,
+        },
+        sortOption: { field: 'Severity', direction: 'desc' },
+    });
+    return `${violationsBasePath}${queryString}`;
+}
+
+export const mostRecentAlertsQuery = gql`
+    query mostRecentAlerts($query: String) {
+        alerts: violations(
+            query: $query
+            pagination: { limit: 3, sortOption: { field: "Violation Time", reversed: true } }
+        ) {
+            id
+            time
+            deployment {
+                clusterName
+                namespace
+                name
+            }
+            policy {
+                name
+                severity
+            }
+        }
+    }
+`;
+
+function ViolationsByPolicySeverity() {
+    const { searchFilter } = useURLSearch();
+    const countsFilterQuery = getRequestQueryStringForSearchFilter(searchFilter);
+    const mostRecentFilterQuery = getRequestQueryStringForSearchFilter({
+        ...searchFilter,
+        Severity: severities.CRITICAL_SEVERITY,
+    });
+    const {
+        data: alertCountData,
+        loading: alertCountLoading,
+        error: alertCountError,
+    } = useAlertGroups(countsFilterQuery);
+    const {
+        data: currentRecentAlertsData,
+        previousData: previousRecentAlertsData,
+        loading: recentAlertsLoading,
+        error: recentAlertsError,
+    } = useQuery<{ alerts: DeploymentAlert[] }>(mostRecentAlertsQuery, {
```

Comment:
Using `DeploymentAlert` as the type here makes me a bit uneasy, as it isn't exactly what is being returned by the gql query even if all of the required fields do exist at runtime. `Partial<DeploymentAlert>` is probably more technically correct but required extensive null checks for properties that we never null.

Reply:
Any reason (asking, not recommending) not to use REST request like:

 /v1/alerts?query=Severity%3ACRITICAL_SEVERITY&pagination.offset=0&pagination.limit=3&pagination.sortOption.field=Violation%20Time&pagination.sortOption.reversed=true

whose response is `ListAlert` which has
* `ListAlertDeployment` https://github.com/stackrox/stackrox/blob/master/proto/storage/alert.proto#L202-L215
* `ListAlertPolicy` https://github.com/stackrox/stackrox/blob/master/proto/storage/alert.proto#L187-L200